server/routes/induction/create/inPrisonWorkCreateController.test.ts

```diff
@@ -0,0 +1,128 @@
+
+  let req: Request
+  let res: Response
+  const next = jest.fn()
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+    req = {
+      session: { prisonerSummary } as SessionData,
+      body: {},
+      user: {} as Express.User,
+      params: { prisonNumber } as Record<string, string>,
+      flash: jest.fn(),
+      path: `/prisoners/${prisonNumber}/create-induction/in-prison-work`,
+    } as unknown as Request
+    res = {
+      redirect: jest.fn(),
+      render: jest.fn(),
+    } as unknown as Response
+  })
+
+  describe('getInPrisonWorkView', () => {
+    it('should get the In Prison Work view given there is no InPrisonWork on the session', async () => {
+      // Given
+      const inductionDto = aShortQuestionSetInductionDto()
+      inductionDto.inPrisonInterests.inPrisonWorkInterests = undefined
+      req.session.inductionDto = inductionDto
+      req.session.inPrisonWorkForm = undefined
+
+      const expectedInPrisonWorkForm: InPrisonWorkForm = {
+        inPrisonWork: [],
+        inPrisonWorkOther: undefined,
+      }
+
+      const expectedView = {
+        prisonerSummary,
+        form: expectedInPrisonWorkForm,
+        backLinkUrl: '/prisoners/A1234BC/create-induction/additional-training',
+        backLinkAriaText: 'Back to Does Jimmy Lightfingers have any other training or vocational qualifications?',
+        errors: noErrors,
+      }
+
+      // When
+      await controller.getInPrisonWorkView(req, res, next)
+
+      // Then
+      expect(res.render).toHaveBeenCalledWith('pages/induction/inPrisonWork/index', expectedView)
+      expect(req.session.inPrisonWorkForm).toBeUndefined()
+      expect(req.session.inductionDto).toEqual(inductionDto)
+    })
+
+    it('should get the In Prison Work view given there is an InPrisonWork already on the session', async () => {
+      // Given
+      const inductionDto = aShortQuestionSetInductionDto()
+      inductionDto.inPrisonInterests.inPrisonWorkInterests = undefined
+      req.session.inductionDto = inductionDto
+
+      const expectedInPrisonWorkForm: InPrisonWorkForm = {
+        inPrisonWork: ['PRISON_LIBRARY', 'WELDING_AND_METALWORK'],
+        inPrisonWorkOther: '',
+      }
+      req.session.inPrisonWorkForm = expectedInPrisonWorkForm
+
+      const expectedView = {
+        prisonerSummary,
+        form: expectedInPrisonWorkForm,
+        backLinkUrl: '/prisoners/A1234BC/create-induction/additional-training',
+        backLinkAriaText: 'Back to Does Jimmy Lightfingers have any other training or vocational qualifications?',
+        errors: noErrors,
+      }
+
+      // When
+      await controller.getInPrisonWorkView(req, res, next)
+
+      // Then
+      expect(res.render).toHaveBeenCalledWith('pages/induction/inPrisonWork/index', expectedView)
+      expect(req.session.inPrisonWorkForm).toBeUndefined()
+      expect(req.session.inductionDto).toEqual(inductionDto)
+    })
+  })
+
+  describe('submitInPrisonWorkForm', () => {
```

Comment:
Does this need a unit test for a successful form submission, or are you going to pick that up in another PR?

Reply:
Doh! Did I not do that? 
Good spot! I'll do that now 👍

Reply:
Done